### PR TITLE
[PERFSCALE-2823]: Self-Managed AWS Loaded Upgrade 4.14 --> 4.15

### PR DESCRIPTION
**Pull request by krishvoor**
This PR allows to perform loaded upgrade on self managed aws 4.14 to 4.15

### CODE REVIEW

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-120nodes-stage-loaded-upgrade415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-249nodes-stage-loaded-upgrade415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-180nodes-stage-loaded-upgrade415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-249nodes-stage-loaded-upgrade415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-120nodes-stage-loaded-upgrade415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-120nodes-stage-loaded-upgrade415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-249nodes-stage-loaded-upgrade415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-180nodes-stage-loaded-upgrade415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-180nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by venkataanil** [comment]
From the log https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/48101/rehearse-48101-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415/1753310676312395776/artifacts/perfscale-aws-120nodes-loaded-upgrade-414to415/workers-scale/build-log.txt we can see that while loop in waitForReady() https://github.com/openshift/release/blob/master/ci-operator/step-registry/workers-scale/workers-scale-commands.sh#L12 is not correctly checking if all the workers moved to Ready state. "oc get nodes" command after that while loop https://github.com/openshift/release/blob/master/ci-operator/step-registry/workers-scale/workers-scale-commands.sh#L33 is still showing many nodes in NotReady state

For example, when I rebooted machine I see "oc get node --output json" showing the status like below.
            {
                "lastHeartbeatTime": "2024-02-02T17:26:25Z",
                "lastTransitionTime": "2024-02-02T17:27:11Z",
                "message": "Kubelet stopped posting node status.",
                "reason": "NodeStatusUnknown",
                "status": "Unknown",
                "type": "Ready"
            }
Though node is not ready in above case, https://github.com/openshift/release/blob/master/ci-operator/step-registry/workers-scale/workers-scale-commands.sh#L12 is considering it as ready as it only check for "type" field. Ideally it should check for both "type" and "status" fields. So the command should be changed to

while [[ $(oc get nodes --no-headers -l node-role.kubernetes.io/worker --output json | jq -r '.items[] | select(.status.conditions[-1].type == "Ready" and .status.conditions[-1].status == "True").status.conditions[-1].type' | wc -w | xargs) != "$1" ]]; do
i.e check for "type==Ready" && "status==True".

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-180nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-180nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-candidate-x86-loaded-upgrade-from-4.14-perfscale-rosa-multiaz-120nodes-stage-loaded-upgrade415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-180nodes-loaded-upgrade-414to415

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml [hunk 1]
```diff
@@ -54,6 +54,90 @@ tests:
...
+    env:
+      BASE_DOMAIN: perfscale.devcluster.openshift.com
+      CHURN: "false"
+      COMPUTE_NODE_TYPE: m5.xlarge
+      CONTROL_PLANE_INSTANCE_TYPE: m5.8xlarge
+      ENABLE_FORCE: "true"
+      GC: "false"
+      ITERATION_MULTIPLIER_ENV: "6"
+      MAX_UNAVAILABLE: "15"
+      PROFILE_TYPE: reporting
```
Comment (krishvoor) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml:74]: Add USER_TAGS

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml [hunk 2]
```diff
@@ -54,6 +54,90 @@ tests:
...
+    env:
+      BASE_DOMAIN: perfscale.devcluster.openshift.com
+      CHURN: "false"
+      COMPUTE_NODE_TYPE: m5.xlarge
+      CONTROL_PLANE_INSTANCE_TYPE: m5.8xlarge
+      ENABLE_FORCE: "true"
+      GC: "false"
+      ITERATION_MULTIPLIER_ENV: "6"
+      MAX_UNAVAILABLE: "22"
+      PROFILE_TYPE: reporting
```
Comment (krishvoor) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml:102]: Add USER_TAGS

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml [hunk 3]
```diff
@@ -54,6 +54,90 @@ tests:
...
+    env:
+      BASE_DOMAIN: perfscale.devcluster.openshift.com
+      CHURN: "false"
+      COMPUTE_NODE_TYPE: m5.xlarge
+      CONTROL_PLANE_INSTANCE_TYPE: m5.8xlarge
+      ENABLE_FORCE: "true"
+      GC: "false"
+      ITERATION_MULTIPLIER_ENV: "6"
+      MAX_UNAVAILABLE: "31"
+      PROFILE_TYPE: reporting
```
Comment (krishvoor) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml:130]: Add USER_TAGS

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-120nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-180nodes-loaded-upgrade-414to415 periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Review by venkataanil — APPROVED** [review]
LGTM

**Comment by svetsa-rh** [comment]
/hold 
@krishvoor Since you are modifying common code in step registry, can you also pls run tests without scale and for 24 nodes 
I want to make sure they are all good with the change.

**Comment by svetsa-rh** [comment]
/hold cancel
I realized that in this case, the common step registry will only affect self-managed clusters. So, we should be good with the change. Just wanted to make sure that we don't affect others tests.

**Comment by krishvoor** [comment]
Minor nit on the way - hence the hold :-)

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-252nodes-loaded-upgrade-414to415

**Comment by venkataanil** [comment]
252 node upgrade succesful. Please merge it :)

**Pull request merged**